lib/streamlit/elements/widgets/button.py

```diff
@@ -512,6 +539,75 @@ def _link_button(
 
         return self.dg._enqueue("link_button", link_button_proto)
 
+    def _page_link(
+        self,
+        page: str,
+        *,  # keyword-only arguments:
+        label: Optional[str] = None,
+        icon: Optional[str] = None,
+        help: Optional[str] = None,
+        disabled: bool = False,
+        use_container_width: Optional[bool] = None,
+    ) -> "DeltaGenerator":
+        page_link_proto = PageLinkProto()
+        page_link_proto.disabled = disabled
+
+        if label is not None:
+            page_link_proto.label = label
+
+        if icon is not None:
+            page_link_proto.icon = validate_emoji(icon)
+
+        if help is not None:
+            page_link_proto.help = dedent(help)
+
+        if use_container_width is not None:
+            page_link_proto.use_container_width = str(use_container_width)
+
+        # Handle external links:
+        if page.startswith("http://") or page.startswith("https://"):
+            if label is None or label == "":
+                raise StreamlitAPIException(
+                    f"The label param is required for external links used with st.page_link - please provide a label."
+                )
+            else:
+                page_link_proto.page = page
+                page_link_proto.external = True
+                return self.dg._enqueue("page_link", page_link_proto)
+
+        ctx = get_script_run_ctx()
+        ctx_main_script = ""
+        if ctx:
+            ctx_main_script = ctx.main_script_path
+
+        main_script_path = os.path.join(os.getcwd(), ctx_main_script)
+        main_script_directory = os.path.dirname(main_script_path)
+
+        # Convenience for handling ./ notation and ensure leading / doesn't refer to root directory
+        page = os.path.normpath(page.strip("/"))
+
+        # Build full path
+        requested_page = os.path.join(main_script_directory, page)
+        all_app_pages = source_util.get_pages(ctx_main_script).values()
+
+        # Handle retrieving the page_script_hash & page
+        for page_data in all_app_pages:
+            full_path = page_data["script_path"]
+            page_name = page_data["page_name"].replace("_", " ").replace("-", " ")
```

Comment:
Just to double-check: do we really want to convert `-` here as well? Isn't it quite a valid character for the display name?